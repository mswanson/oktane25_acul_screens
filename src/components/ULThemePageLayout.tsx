import type { HTMLAttributes, ReactNode } from "react";

import { cn } from "@/lib/utils";

export interface ULThemePageLayoutProps extends HTMLAttributes<HTMLDivElement> {
  /**
   * The content of the layout.
   */
  children: ReactNode;
  /**
   * Optional class names for additional styling or overriding default styles.
   */
  className?: string;
}

const ULThemePageLayout = ({
  children,
  className,
  ...rest
}: ULThemePageLayoutProps) => {
  const themedStyles =
    "flex px-10 py-20 justify-page-layout bg-background-page bg-(image:--ul-theme-page-bg-background-image-url)";
  return (
    <div className={cn(themedStyles, className)} {...rest}>
      {children}
    </div>
  );
};

export default ULThemePageLayout;
