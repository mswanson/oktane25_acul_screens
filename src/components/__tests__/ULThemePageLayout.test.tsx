import { render } from "@testing-library/react";

import ULThemePageLayout from "@/components/ULThemePageLayout";

describe("ULThemePageLayout Component Test Suite", () => {
  // Snapshot Tests
  it("matches snapshot with default props", () => {
    const { container } = render(
      <ULThemePageLayout>
        <div>Test Content</div>
      </ULThemePageLayout>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("matches snapshot with custom className", () => {
    const { container } = render(
      <ULThemePageLayout className="custom-class">
        <div>Test Content</div>
      </ULThemePageLayout>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("matches snapshot with additional props", () => {
    const { container } = render(
      <ULThemePageLayout id="test-id" data-testid="test-layout">
        <div>Test Content</div>
      </ULThemePageLayout>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  // Functional Tests
  it("renders children correctly", () => {
    const { getByText } = render(
      <ULThemePageLayout>
        <div>Test Content</div>
      </ULThemePageLayout>
    );

    expect(getByText("Test Content")).toBeInTheDocument();
  });

  it("applies default themed styles", () => {
    const { container } = render(
      <ULThemePageLayout>Test Content</ULThemePageLayout>
    );

    expect(container.firstChild).toHaveClass(
      "flex px-10 py-20 justify-page-layout bg-background-page bg-(image:--ul-theme-page-bg-background-image-url)"
    );
  });

  it("applies custom className", () => {
    const { container } = render(
      <ULThemePageLayout className="custom-class">
        Test Content
      </ULThemePageLayout>
    );

    expect(container.firstChild).toHaveClass("custom-class");
  });

  it("passes additional props to the root element", () => {
    const { container } = render(
      <ULThemePageLayout id="test-id" data-testid="test-layout">
        Test Content
      </ULThemePageLayout>
    );

    expect(container.firstChild).toHaveAttribute("id", "test-id");
    expect(container.firstChild).toHaveAttribute("data-testid", "test-layout");
  });
});
