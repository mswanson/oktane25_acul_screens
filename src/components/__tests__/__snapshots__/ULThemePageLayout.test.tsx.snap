// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ULThemePageLayout Component Test Suite matches snapshot with additional props 1`] = `
<div
  class="flex px-10 py-20 justify-page-layout bg-background-page bg-(image:--ul-theme-page-bg-background-image-url)"
  data-testid="test-layout"
  id="test-id"
>
  <div>
    Test Content
  </div>
</div>
`;

exports[`ULThemePageLayout Component Test Suite matches snapshot with custom className 1`] = `
<div
  class="flex px-10 py-20 justify-page-layout bg-background-page bg-(image:--ul-theme-page-bg-background-image-url) custom-class"
>
  <div>
    Test Content
  </div>
</div>
`;

exports[`ULThemePageLayout Component Test Suite matches snapshot with default props 1`] = `
<div
  class="flex px-10 py-20 justify-page-layout bg-background-page bg-(image:--ul-theme-page-bg-background-image-url)"
>
  <div>
    Test Content
  </div>
</div>
`;
